### PR TITLE
[PUBDEV-5507] More robust check for colorama  version

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

File: h2o-py/h2o/backend/__init__.py [hunk 1]
```diff
@@ -48,10 +48,13 @@
 from distutils.version import StrictVersion
 import sys
 
-if StrictVersion(colorama.__version__) < StrictVersion("0.3.8"):
+if not hasattr(colorama, '__version__'):
+    print("[WARNING] H2O requires colorama module of version 0.3.8 or newer. You have older version .\n"
```
Comment (mmalohlava) [inline h2o-py/h2o/backend/__init__.py]: Should we fail directly?

btw: in theory, setup.py for H2O python package should require version >= 0.3.8.

**Comment by michalkurka** [comment]
@mmalohlava Our setup.py already has the dependency:

    install_requires=["requests", "tabulate", "future", "colorama>=0.3.8"],

When does the issue happen?

File: h2o-py/h2o/backend/__init__.py (hunk 1 above)
Comment (jakubhava) [reply h2o-py/h2o/backend/__init__.py]: It is already there

**Comment by jakubhava** [comment]
We already have setup.py requirement specified. That means that you are only allowed to build and install H2O if you have colorama 0.3.8+.

I fails if you change the dependency afterwards to colorama 0.3.7 and bellow. Then I think about it, the core issue is PySparkling in this case. It requires `colorama` because H2O needs it, but we bundle H2O python files inside. The solution there is to specify the requirement here as well https://github.com/h2oai/sparkling-water/blob/f52ed7312ba371d5e769435655289730d1385da2/py/setup.py#L49

If we want to handle the cases when someone changes the colorama dependency after H2O is already installed, then this change has purpose I think, otherwise no.

**Comment by mmalohlava** [comment]
Sorry for blocking this PR, but setup.py needs proper install_requirements with version boundaries:

https://packaging.python.org/discussions/install-requires-vs-requirements/

See:
image

**Comment by jakubhava** [comment]
@mmalohlava no worries. So you would like to extend this PR to handle properly also other libraries - `"requests", "tabulate", "future"` ? It is already done for colorama

**Comment by mmalohlava** [comment]
@michalkurka wdyt?

**Comment by jakubhava** [comment]
just second note: I'm all for that, I would just do it in scope of different change as we know the colorama fixed version will be fine, we need to test the other ones

**Pull request merged**